### PR TITLE
Detect phrases in french raster layer names as hints that a layer represents an elevation surface

### PR DESCRIPTION
**Pull request by nicogodet**
- `mnt` stand for `dtm`
- `mns` for `dsm`
- `rge` -> https://geoservices.ign.fr/rgealti
- `alti` for `elev`

We (well, me in my work) use `sl` for "Surface Libre" which is the free surface elevation (mainly computed using flood model). 
But I'm afraid this will detect too many false positive.
I wonder if this list could be exposed to user in settings

### CODE REVIEW

File: src/core/raster/qgsrasterlayerelevationproperties.cpp [hunk 1]
```diff
@@ -241,7 +241,12 @@ bool QgsRasterLayerElevationProperties::layerLooksLikeDem( QgsRasterLayer *layer
   static const QStringList sPartialCandidates{ QStringLiteral( "dem" ),
       QStringLiteral( "height" ),
       QStringLiteral( "elev" ),
-      QStringLiteral( "srtm" ) };
+      QStringLiteral( "srtm" ),
+      // French hints
```
Comment (pathmapper) [inline src/core/raster/qgsrasterlayerelevationproperties.cpp:245]:
```suggestion
      // French hints
      QStringLiteral( "mne" ),
```

Comment (pathmapper) [reply src/core/raster/qgsrasterlayerelevationproperties.cpp:245]: @nicogodet not sure if didn't include this intentionally.

Seems to be used, e.g. http://eductice.ens-lyon.fr/EducTice/recherche/geomatique/veille/imagerie-3d/MNT/index.htm but you're the native speaker :-)

Comment (nicogodet) [reply src/core/raster/qgsrasterlayerelevationproperties.cpp:245]: MNE = MNS but you are right, both can be used !

**Comment by lbartoletti** [comment]
> I wonder if this list could be exposed to user in settings

if it's intended to have hints for each language, they could be in the settings/text files, couldn't they?

**Comment by nicogodet** [comment]
IMO it would be better than hardcoded values (or in addition to generic english hardcoded values).

**Comment by nyalldawson** [comment]
There's an explanation at https://github.com/qgis/QGIS/pull/30245 why this list should NOT be localised

**Comment by nicogodet** [comment]
> but adding hardcoded localized variants of the strings is encouraged

Did I miss-interpreted this comment ?

**Comment by nyalldawson** [comment]
@nicogodet

No, your approach here is correct. I'm referring to the comment regarding having a per language list.

**Review by nyalldawson — APPROVED** [review]

**Comment by nicogodet** [comment]
Personally, I suggested to expose to user in GUI a way to add user defined hints

Roughly: 
```c++
static const QStringList sPartialCandidates{ QStringLiteral( "dem" ),
      QStringLiteral( "height" ),
      QStringLiteral( "elev" ),
      QStringLiteral( "srtm" ),
      // French hints
      QStringLiteral( "mne" ),
      QStringLiteral( "mnt" ),
      QStringLiteral( "mns" ),
      QStringLiteral( "rge" ),
      QStringLiteral( "alti" ) };
sPartialCandidates << settings->userDemHints();
```

**Pull request merged**